pyseir/rt/infer_rt.py

```diff
@@ -0,0 +1,617 @@
+            numerator = likelihoods[current_day] * current_prior
+
+            # (5c) Calculate the denominator of Bayes' Rule P(k)
+            denominator = np.sum(numerator)
+
+            # Execute full Bayes' Rule
+            if denominator == 0:
+                # Restart the baysian learning for the remaining series.
+                # This is necessary since otherwise NaN values
+                # will be inferred for all future days, after seeing
+                # a single (smoothed) zero value.
+                #
+                # We understand that restarting the posteriors with the
+                # re-initial prior may incur a start-up artifact as the posterior
+                # restabilizes, but we believe it's the current best
+                # solution for municipalities that have smoothed cases and
+                # deaths that dip down to zero, but then start to increase
+                # again.
+
+                posteriors[current_day] = reinit_prior
+            else:
+                posteriors[current_day] = numerator / denominator
+
+            # Monitors if posterior is lagging excessively behind signal in likelihood
+            # TODO future can return cumulative lag and use to scale sigma up only when needed
+            monitor.evaluate_lag_using_argmaxes(
+                current_day=loop_idx,
+                current_sigma=current_sigma,
+                prev_post_am=posteriors[previous_day].argmax(),
+                prior_am=current_prior.argmax(),
+                like_am=likelihoods[current_day].argmax(),
+                post_am=numerator.argmax(),
+            )
+
+            # Add to the running sum of log likelihoods
+            log_likelihood += np.log(denominator)
+            loop_idx += 1
+
+        self.log_likelihood = log_likelihood
+
+        if plot:
+            plotting.plot_posteriors(x=posteriors)  # Returns Figure.
+            # The interpreter will handle this as it sees fit. Normal builds never call plot flag.
+
+        start_idx = -len(posteriors.columns)
+
+        return dates[start_idx:], posteriors, start_idx
+
+    def get_available_timeseries(self):
+        """
+        Determine available timeseries for Rt inference calculation 
+        with constraints below.
+
+
+        Returns
+        -------
+        available_timeseries: 
+          array of available timeseries saved as TimeseriesType
+        """
+        available_timeseries = []
+        _, cases = self.get_timeseries(TimeseriesType.NEW_CASES.value)
+        _, deaths = self.get_timeseries(TimeseriesType.NEW_DEATHS.value)
+
+        if np.sum(cases) > self.min_cases:
+            available_timeseries.append(TimeseriesType.NEW_CASES)
+
+        if np.sum(deaths) > self.min_deaths:
+            available_timeseries.append(TimeseriesType.NEW_DEATHS)
+
+        return available_timeseries
+
+    def infer_all(self, plot=True, shift_deaths=0):
```

Comment:
My ideal way to call this function would be:
```
def infer_all(self, timeseries: List[pd.Series], plot=True, shift_deaths=0):
```
or maybe
```
def infer_all(self, fips: str, timeseries: List[pd.Series], plot=True, shift_deaths=0):
```

that way it frees a lot of the work from the class state and easier to see how timeseries are passed through.